src/agentscope/agents/rag_agents.py

```diff
@@ -0,0 +1,339 @@
+# -*- coding: utf-8 -*-
+"""
+This example shows how to build an agent with RAG
+(with LlamaIndex or Langchain)
+"""
+
+from abc import ABC, abstractmethod
+from typing import Optional, Any
+import importlib
+from loguru import logger
+
+
+from agentscope.agents.agent import AgentBase
+from agentscope.message import Msg
+from agentscope.prompt import PromptEngine
+from agentscope.models import load_model_by_config_name
+
+try:
+    from agentscope.rag import RAGBase, LlamaIndexRAG
```

Comment:
Catching the exceptions within `RAGBase` and `LlamaIndexRAG` file is recommended.

Reply:
Moved rag agents to example for now as discussed, so the problem does not exist now.